tools/wptrunner/wptrunner/browsers/wktr.py

```diff
@@ -0,0 +1,239 @@
+    pass
+
+
+def browser_kwargs(logger, test_type, run_info_data, config, **kwargs):
+    args = list(kwargs["binary_args"])
+
+    args.append("--allow-any-certificate-for-allowed-hosts")
+
+    for host in config.domains_set:
+        args.append('--allowed-host')
+        args.append(host)
+
+    args.append("-")
+
+    return {"binary": kwargs["binary"],
+            "binary_args": args}
+
+
+def executor_kwargs(logger, test_type, test_environment, run_info_data,
+                    **kwargs):
+    executor_kwargs = base_executor_kwargs(test_type, test_environment, run_info_data,
+                                           **kwargs)
+    return executor_kwargs
+
+
+def env_extras(**kwargs):
+    return []
+
+
+def env_options():
+    return {"server_host": "127.0.0.1",
+            "testharnessreport": "testharnessreport-wktr.js"}
+
+
+def update_properties():
+    return (["debug", "os", "processor"], {"os": ["version"], "processor": ["bits"]})
+
+
+class WKTRBrowser(Browser):
+    """Class that represents an instance of WebKitTestRunner.
+
+    Upon startup, the stdout, stderr, and stdin pipes of the underlying WebKitTestRunner
+    process are connected to multiprocessing Queues so that the runner process can
+    interact with WebKitTestRunner through its protocol mode.
+    """
+
+    def __init__(self, logger, binary="WebKitTestRunner", binary_args=[], **kwargs):
+        super().__init__(logger)
+
+        self._args = [binary] + binary_args
+        self._proc = None
+
+        build_root_path = os.path.dirname(binary)
+
+        def append_env(var, item, separator):
+            raw = os.environ.get(var)
+            old = raw.split(separator) if raw is not None else []
+            return separator.join(old + [item])
+
+        env = {}
+
+        if sys.platform.startswith("darwin"):
+            env["CA_DISABLE_GENERIC_SHADERS"] = "1"
+            env["__XPC_CA_DISABLE_GENERIC_SHADERS"] = "1"
+
+            env["DYLD_LIBRARY_PATH"] = append_env(
+                "DYLD_LIBRARY_PATH", build_root_path, ":"
+            )
+            env["__XPC_DYLD_LIBRARY_PATH"] = append_env(
+                "__XPC_DYLD_LIBRARY_PATH", build_root_path, ":"
+            )
+            env["DYLD_FRAMEWORK_PATH"] = append_env(
+                "DYLD_FRAMEWORK_PATH", build_root_path, ":"
+            )
+            env["__XPC_DYLD_FRAMEWORK_PATH"] = append_env(
+                "__XPC_DYLD_FRAMEWORK_PATH", build_root_path, ":"
+            )
+
+            env["SQLITE_EXEMPT_PATH_FROM_VNODE_GUARDS"] = "/"
+            env["__XPC_SQLITE_EXEMPT_PATH_FROM_VNODE_GUARDS"] = "/"
+
+        self._extra_env = env
+
+    def start(self, group_metadata, **kwargs):
+        self.logger.debug("Starting WebKitTestRunner: %s..." % self._args[0])
+        #self.logger.debug(repr(self._args))
+
+        env = os.environ
+        env.update(self._extra_env)
+
+        # Unfortunately we need to use the Process class directly because we do not
```

Comment:
Is there a reason not to just use `subprocess` in that case?

Reply:
This is mostly copied from `content_shell`, honestly. I _assumed_ there was a reason!